src/Microsoft.ML.TimeSeries/IidChangePointDetector.cs

```diff
@@ -112,30 +111,30 @@ private static IDataTransform Create(IHostEnvironment env, Arguments args, IData
             return new IidChangePointDetector(env, args).MakeDataTransform(input);
         }
 
-        internal override IStatefulTransformer Clone()
+        IStatefulTransformer IStatefulTransformer.Clone()
         {
             var clone = (IidChangePointDetector)MemberwiseClone();
-            clone.StateRef = (State)clone.StateRef.Clone();
-            clone.StateRef.InitState(clone, Host);
+            clone.Base.StateRef = (IidAnomalyDetectionBase.State)clone.Base.StateRef.Clone();
```

Comment:
Not sure if you want to merge `clone.Base.StateRef =` and `clone.Base.StateRef.InitState` by using constructor.

Reply:
Please see my previous comment on this. 

---
In reply to: [254482306](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254482306) [](ancestors = 254482306)